src/auth/hooks/useAuthProvider.ts

```diff
@@ -56,16 +57,24 @@ export function useAuthProvider({
   }, [authenticating]);
 
   useEffect(() => {
-    if (!authenticated && !authenticating) {
+    if (!authenticated && !authenticating && permitCredentialsAPI) {
+      setPermitCredentialsAPI(false);
       loginWithCredentialsManagementAPI(handleLogin);
     }
   }, [authenticated, authenticating]);
 
   const userDetails = useQuery<UserDetails>(userDetailsQuery, {
     client: apolloClient,
-    skip: !authenticated
+    skip: !authenticated,
+    fetchPolicy: "cache-and-network"
```

Comment:
Shouldn't this be the default "network-only"? Do we ever fetch this from cache?

Reply:
Theoretically, the cache could be used on page refresh, this is why I left "cache-and-network".

Reply:
Apollo cache is only there for the duration of runtime, it's gone when you do a full page refersh (it's not persisted in local/session storage)

Reply:
However, change again to "cache-and-network" might be needed in https://github.com/saleor/saleor-dashboard/pull/1792,